applications/CompressiblePotentialFlowApplication/python_scripts/apply_far_field_process.py

```diff
@@ -71,6 +72,9 @@ def Execute(self):
             if(tmp < pos+1e-9):
                 node.Fix(CompressiblePotentialFlowApplication.VELOCITY_POTENTIAL)
                 node.SetSolutionStepValue(CompressiblePotentialFlowApplication.VELOCITY_POTENTIAL,0,self.inlet_phi)
+                if self.is_adjoint:
+                    node.Fix(CompressiblePotentialFlowApplication.ADJOINT_VELOCITY_POTENTIAL)
```

Comment:
you could check if the Modelpart has `ADJOINT_VELOCITY_POTENTIAL`as SolutionStepVariable rather than specifying it with a flag
=> this way you could skip the flag

Reply:
Great!